### ISSUE TITLE
Problems with latest change in previewing

### ISSUE THREAD

**Issue by GitHubRulesOK** [sumatrapdfreader/sumatrapdf#2507]
After recent update  I dont seem to be able to restore previewing PDF even after downgrade to older version,
having to consider registry clean or restore from pre recent installs on 32bit win7 and 64bit win 11

![image](https://user-images.githubusercontent.com/12754913/165008051-99978268-2c19-41a0-b125-1adf8c55b0bb.png)

cleaning registry on win7 and reinstall 3.4.14482 works and shows preview 
HOWEVER updating to 3.4.14484 preview fails before and after SumatraPDF is assigned as default PDF handler

**Comment by kjk** [comment]
In retrospect I should have made those changes on a branch and not yet publish in pre-release.

Those are still changes in progress and untested.

Were you installing for all users or just for current user?

The difference is that when installing for all users, we write to HKLM registry root and for current user to HKCU.

In current / past SumatraPDF, we would just write to both.

I've updated the installer to new logic but not yet the installer for preview / search plugins, so that probably has something to do with it.

Not even sure if can install preview / search unless installing for all users, must test.

**Comment by GitHubRulesOK** [comment]
tried different setups for single in program files and local appdata but I am admin level on windows 11 untested as standard user on 10 as dont want to mess that one but win 7 is also elevated user so perhaps thats part of it.

**Comment by kjk** [comment]
How are you testing previewing is working?

Just  using File Explorer's thumbnail view in a folder with PDF files?

**Comment by GitHubRulesOK** [comment]
yes as above previewer pane is set on in most folders
for a while there was an issue that during install the focused file would tend to block previewer install but more recently that issue seemed to have been addressed or i was more diligent to not have dozens of files open or showing in previewer. when I see a problem arise I tend to close explorer even on occasion using task mgr to kill any process

**Comment by GitHubRulesOK** [comment]
OK unsure where the problems were but certainly currently bithout having reset the sandbox (thus no restart etc)
with install in local appdata in sandbox it is working as version  3.4.14485 showing PDF in preview pane
![image](https://user-images.githubusercontent.com/12754913/165508237-0799d816-fed0-4845-bfae-d10d88574e90.png)

**Comment by GitHubRulesOK** [comment]
@kjk 
taking current installer I am currently inside sandbox and batch run the installer but as single admin user thus offers local appdata I add option to preview and installer warns preview dll was not correct (but had unfocused preview EXCEPT the CMD was showing in previewer) anyway dont know if the log helps
[sumatra-install-log.txt](https://github.com/sumatrapdfreader/sumatrapdf/files/8578434/sumatra-install-log.txt)

**Comment by GitHubRulesOK** [comment]
Sorry for heavy download but ran the script outside the sandbox to duplicate my method
Outside sandbox thus more conventional use of installer all is swell there is no complaint if i ensure nothing is previewed except the script and thumbnails appear as well as previewer pane, thus I think the image support inside sandbox is the thumbnail problem but still not clear as to why previewer install grumbles unless its related.

So in summary looks like the installer is stable for normal users just not those that try full features in a VM

**Comment by kjk** [comment]
Yeah, I'm going to say that preview (and probably search) won't work in sandbox.

It seems to work in a regular setup.

**Comment by GitHubRulesOK** [comment]
Ok here we go 3.4.14502
On Win7 installer looks bigger but that's OK its pre-rel
![image](https://user-images.githubusercontent.com/12754913/165746583-0bf4cc58-9167-4bbb-b51d-8dea88b83f03.png)

However latest update changes all pdf icons to lovely green epub ones
![image](https://user-images.githubusercontent.com/12754913/165746690-2e4ce386-bc15-465f-b02d-73e9434f493e.png)

**Comment by kjk** [comment]
re: green icon. Was it a clean win 7 install or was there prior installation?
Either way, I hopefully fixed it.

**Comment by GitHubRulesOK** [comment]
win 7 unit uses check for update and with no browser active should see updates afresh but it took 3 seperate attempts to see the increase from 3.4.14502 to 14504 ! but again it s  a slow device (is there some time out ? )

Icons still green ebub but could be cached, what is noticeable is xps preview is not working but was before (via MSXPS?)
killed explorer to see if that helps restarted and xps is working again with zoom in/out
epub has mupdf icon and different files have two types of yellow pdf so manual association show the smaller icon in mid doc but associated by SumatraPDF show the large watchman style
clicking djvu crashes preview handler surrogate host  but that could be me not rebooting cleanly, and there was a system file related update so starting that unit again !

**Comment by GitHubRulesOK** [comment]
Not sure exactly why I needed to re-trigger assignment but on Win7x32 the Icons now have per your recent changes reverted to Yellow for PDF so looks like icon cache was holding back (this is the slowest of working units as in need of a good refresh)

@kjk At this point I think the OP is probably addressed perhaps close this and open any fresh issue as new? but what can I expect for extended previews ?
